geomstats/_backend/numpy/__init__.py

```diff
@@ -93,6 +93,25 @@
 from . import random  # NOQA
 from .common import to_ndarray  # NOQA
 
+DTYPES = [int32, int64, float32, float64]
+
+
+def to_numpy(x):
+    return x
+
+
+def convert_to_wider_dtype(tensor_list):
+    dtype_list = [x.dtype for x in tensor_list]
```

Comment:
`DTYPES` could be a dict with keys int32 etc, and values 1..4 and here you could have
`dtype_list = [DTYPES[x.dtype] for x in tensor_list]`
so that the `wider_dtype_index` is just `max(dtype_list)` and you don't need to look for it by hand below

Reply:
Done.